lib/internal/buffer.js

```diff
@@ -960,6 +1068,10 @@ function writeFloatBackwards(val, offset = 0) {
   return offset;
 }
 
+function reset() {
+  this.readerIndex = 0;
+}
```

Comment:
Maybe this could be a seek function? one can set any index.

```suggestion
function seek(n) {
  this.readerIndex = n;
}
```